python/src/scipp/compat/mantid.py

```diff
@@ -26,8 +26,46 @@ def convert_instrument(ws):
     return sc.Variable(value=compInfo)
 
 
-def init_pos_spectrum_no(nHist, ws):
-
+def make_detector_info(ws):
+    nHist = ws.getNumberHistograms()
+    det_info = ws.detectorInfo()
+    # det -> spec mapping
+    nDet = det_info.size()
+    spectrum = sc.Variable([sc.Dim.Detector],
+                           shape=(nDet, ),
+                           dtype=sc.dtype.int32)
+    has_spectrum = sc.Variable([sc.Dim.Detector],
+                               values=np.full((nDet, ), False))
+    spectrum_ = spectrum.values
+    has_spectrum_ = has_spectrum.values
+    spec_info = ws.spectrumInfo()
+    for i in range(nHist):
```

Comment:
Minor suggested improvement...

Iterators now supported. Should be able to do:
```
for spec in spec_info:
    ...
```